libs/shared/storage/src/lib/simulation-storage.service.ts

```diff
@@ -69,8 +80,19 @@ export class SimulationStorageService {
   }
 
   public async deleteSimulationArchive(runId: string): Promise<void> {
-    await this.storage.deleteObject(
-      this.endpoints.getSimulationRunCombineArchiveS3Path(runId),
-    );
+    const fileObject = this.endpoints.getSimulationRunCombineArchiveS3Path(runId);
+    await this.storage.deleteObject(fileObject);
+
+    await this.storage.getObject(fileObject)
+      .then((file: AWS.S3.GetObjectOutput): void => {
+        if (!file.DeleteMarker) {
+          throw new InternalServerErrorException(`COMBINE archive could not be deleted for simulation run '{runId}'.`);
+        }
+      })
+      .catch((error: any): void => {
+        if (error.status !== HttpStatus.NOT_FOUND && error.statusCode !== HttpStatus.NOT_FOUND) {
```

Comment:
This won't ever catch since S3 will not throw an error. Instead, the s3GetObjectOutput type will contain an error that has information about the error that occurred. Similarly, there won't be standard http status codes to test for here

Reply:
This is intended to catch files which were deleted (404 code). This seemed to work in my testing.

Reply:
That confuses me, since if the delete marker is set as above, then the object itself still exists, though its versions may have been deleted. This means we should never get a 404.  if we are getting a 404, then the the file should always be null

Reply:
With out bucket, after deletion, `getObject` raises an error with `statusCode` `404` and `code` `NoSuchKey`. The if statement in the catch catches these errors and re-throws all other errors.

The code was written to handle both cases when the bucket is and isn't versioned. For clarity, I removed the DeleteMarker check.